### PR TITLE
[TechDebt]--Rename "requiresLighting"/"requires_lighting" to "forceFlatShading"/"force_flat_shading"

### PR DESCRIPTION
**Pull request by jturner65**
## Motivation and Context
This PR renames "requiresLighting" (in C++)/"requires_lighting"(in python) to "forceFlatShading" (in C++)/"force_flat_shading"(in python).  Despite the original name, this field is only used to determine whether or not flat shading should be forcibly applied to an asset regardless of its material or other specifications, and so this field is being renamed (and values being inverted, where appropriate) to more accurately reflect its purpose.

This change will necessitate a single-line [change to Habitat-lab here](https://github.com/facebookresearch/habitat-lab/blob/598f64813b4e5910d7e00dd07cb304dc9620eb02/habitat/tasks/rearrange/obj_loaders.py#L30) but this change is a non-critical one, since the value being specified for "requires_lighting" is the default value for the attributes.    [Here is the Habitat-lab refactor PR.](https://github.com/facebookresearch/habitat-lab/pull/731)

## How Has This Been Tested
Locally C++ and python tests.

## Types of changes

- [x] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by Skylion007** [comment]
Won't the HM3D Dataset need to be updated?

**Comment by jturner65** [comment]
> 
> 
> Won't the HM3D Dataset need to be updated?

Nope, they shouldn't, the configs for HM3D do not reference that field.

File: data/test_assets/dataset_tests/dataset_1/objects/object_0/dataset_test_object1_0.object_config.json [hunk 1]
```diff
@@ -8,7 +8,7 @@
 	"friction_coefficient": 0.5,
 	"restitution_coefficient": 0.2,
 	"units_to_meters":1.0,
-	"requires_lighting": true,
+  "force_flat_shading": false,
```
Comment (Skylion007) [inline data/test_assets/dataset_tests/dataset_1/objects/object_0/dataset_test_object1_0.object_config.json:11]: Indentation is weird. Is this find/replace error?

**Review by aclegg3 — APPROVED** [review]
Breaking change?
Some user code may need to do this migration.

**Pull request merged**